Week3/nodejs_tasks/routes/tasks.js

```diff
@@ -0,0 +1,150 @@
+const express = require('express');
+const app = express();
+
+app.get('/', (req, res, next) => {
+    req.getConnection((error, conn) => {
+        conn.query("select * FROM tbl_tasks ORDER BY id DESC", (err, rows, fields) => {
+            if (err) {
+                req.flash('error', err)
+                res.render('task/list', {
+                    title: 'Task List',
+                    data: ' '
+                })
+            }else{
+                res.render('task/list', {
+                    title:'Task List',
+                    data:rows
+                })
+            }
+        })
+    })
+})
+
+app.get('/add', (req, res, next) => {
+    res.render('task/add', {
+        title:'Add New Task',
+        task_name:' '
+    })
+})
+
+app.post('/add', (req, res, next) => {
+    req.assert('task_name', 'Name is required').noEmpty();
+
+    let errors = req.validationErrors();
+
+    if(!errors){
+        let task = {
+            task_name: req.sanitize('task_name').escape().trim()
+        }
+
+        req.getConnection((error, conn) => {
+            conn.query("INSERT INTO tbl_tasks SET ? ", task, (err, result) => {
+                if(err){
+                    req.flash('error', err)
+                    res.render('task/add', {
+                        title: 'Add New Task',
+                        task_name: task.task_name
+                    })
+                }else{
+                    req.flash('success', 'Data Added successfully!');
+                    res.render('task/add', {
+                        title: 'Add  New User',
+                        task_name: ' '
+                    })
+                }
+            })
+        })
+    }else{
+        let error_msg = '';
+        errors.forEach((error) => {
+            error_msg += error_msg + '<br/>'
+        })
+
+        req.flash('error', error_msg);
+        res.render('user/add', {
+            title: 'Add New User',
+            task_name: req.body.task_name
+        })
+    }
+})
+
+app.get('/edit/(:id)', (req, res, next) => {
+    req.getConnection((error, conn) => {
+        conn.query("SELECT * FROM tbl_tasks WHERE id = " +
```

Comment:
Be careful here you are not using prepared statement and string injection. You should not insert any values given by a call on your api even if it's your front end, you could vulnerable to attack.